src/scipp/testing/assertions.py

```diff
@@ -57,102 +77,103 @@ def assert_identical(a: T, b: T) -> None:
     AssertionError
         If the objects are not identical.
     """
-    try:
-        _assert_identical_impl(a, b)
-    except AssertionError as exc:
-        if hasattr(exc, '__notes__'):
-            # See comment above.
-            notes = []
-            rest = -1
-            for i, note in enumerate(exc.__notes__):
-                if 'PREPOSITION' in note:
-                    notes.append(note.replace('PREPOSITION', 'in'))
-                    rest = i
-                    break
-            notes.extend(
-                note.replace('PREPOSITION', 'of') for note in exc.__notes__[rest + 1 :]
-            )
-            exc.__notes__ = notes
-        raise
-
-
-def _assert_identical_impl(a: T, b: T) -> None:
-    assert type(a) == type(b)
-    if isinstance(a, Variable):
-        _assert_identical_variable(a, b)
-    elif isinstance(a, DataArray):
-        _assert_identical_data_array(a, b)
-    elif isinstance(a, Dataset):
-        _assert_identical_dataset(a, b)
-    elif isinstance(a, DataGroup):
-        _assert_identical_datagroup(a, b)
-    else:
-        assert a == b
-
-
-def _assert_identical_variable(a: Variable, b: Variable) -> None:
-    assert a.sizes == b.sizes
-    assert a.unit == b.unit
-    assert a.dtype == b.dtype
-    assert (a.bins is None) == (b.bins is None)
-    if a.bins is None:
-        _assert_identical_dense_variable_data(a, b)
-    else:
-        _assert_identical_binned_variable_data(a, b)
-
-
-def _assert_identical_binned_variable_data(a: Variable, b: Variable) -> None:
-    # Support for iterating over bin contents is limited in Python.
-    # So, simply use `identical` even though it does not produce good error messages.
-    assert a.bins.unit == b.bins.unit
-    assert identical(a, b)
-
-
-def _assert_identical_dense_variable_data(a: Variable, b: Variable) -> None:
-    with _add_note('values'):
-        np.testing.assert_array_equal(
-            a.values, b.values, err_msg='when comparing values'
-        )
-    if a.variances is not None:
-        assert b.variances is not None, 'a has variances but b does not'
-        with _add_note('variances'):
-            np.testing.assert_array_equal(
-                a.variances, b.variances, err_msg='when comparing variances'
+    _AssertSimilar(np.testing.assert_array_equal).assert_similar(a, b)
+
+
+class _AssertSimilar:
```

Comment:
I want the individual methods to use the same version of  `assert_numerical_similarity`.
If they were free functions the similarity function would have to be passed from function to function as an extra argument, that seemed more cumbersome than just making it into a class.

Reply:
Well, you have to pass `self` as an argument now. That's no less cumbersome IMHO.

Reply:
I think the class version is a bit less noisy but I can change it to the free function version if you think that is better.

Reply:
I find it simpler to not use a class in this way because with the class, you need to read it completely to understand that it is only used for passing parameters and does not encode additional state.